### DISCUSSION TITLE
Vite build / deployment problem

### DISCUSSION THREAD

**Discussion by danielbayley80** [sveltejs/kit#11272] [General]
This is likely a noob question, and might be a vite thing or a sveltekit, or my lack of understanding not sure ..... but I haven't been able to GPT my way out of it.

I run "vite build" and everything goes okay. Then it appears to try and run my app (not what I expected) and errors out because there is no production connection string locally.

Is this sveltekit trying to pre-render SSR content? I was just hoping to get a standard vite build output which I push to the server.

What am I missing here?

``
✓ 459 modules transformed.
Environment : production

node:internal/event_target:1037
  process.nextTick(() => { throw err; });
                           ^
Error [MongoParseError]: Invalid scheme, expected connection string to start with "mongodb://" or "mongodb+srv://"
``

**Comment by danielbayley80** [comment]
okay, if in doubt .... read F'in manual ....

"SvelteKit will load your +page/layout(.server).js files (and all files they import) for analysis during the build. Any code that should not be executed at this stage must check that building from [$app/environment](https://kit.svelte.dev/docs/modules#$app-environment) is false:"

Most of my app depends on mongodb, so I will likely need to put this build check high up in the stack. What is the implication of svelte not running some / all of your app. Is this mainly a pre-rendering thing?

Edit ... it's not clear to me if I should use my dev connection string, or just prevent database code from running.

**Reply by alexbjorlig** [reply]
There are many strategies when it comes to [Static site generation](https://kit.svelte.dev/docs/adapter-static) and databases.

Do you need static site generation for your app? If no, then you just have to disable it. If yes, then I can help clarify any understanding if you share some more details about what you are trying to build :)

**Reply by danielbayley80** [reply]
Thank. I decided to disable it across the board. 
Out of luck I separated my marketing / CMS site from my app when I started using Sveltekit for the app so it's not a big deal and I'm not missing out on a lot by disabling it. I should have read deeper in to the documentation as there are likely some key design decisions I would have done differently. I guess one solution would be to have a build server with access to the production data.

**Reply by alexbjorlig** [reply]
Yes, it's a very common approach to build against the production source. If you need help in the future, you can ping me if needed :)
